src/NodaTime.Demo/DateIntervalDemo.cs

```diff
@@ -69,5 +69,20 @@ public void Contains_Interval()
             var result = Snippet.For(interval.Contains(june));
             Assert.AreEqual(true, result);
         }
+
+        [Test]
+        public void Deconstruction()
+        {
+            var calendar = CalendarSystem.Gregorian;
+            LocalDate start = new LocalDate(2017, 1, 1);
+            LocalDate end = new LocalDate(2017, 12, 31);
+
+            DateInterval value = new DateInterval(start, end);
+
+            value.Deconstruct(out LocalDate actualStart, out LocalDate actualEnd);
```

Comment:
Okay, I've now merged the new Snippet code. If you *add*

```csharp
Snippet.SilentForAction(() => value.Deconstruct(out var _, out var _));
```

... it should be fine. The idea is that it's "silent" because the whole statement is removed. We can't just remove the `Snippet.For` part, because of the way definite assignment works. Let me know if you need more details.

Reply:
Great, thanks! Generated the snippet locally and this is the new generated snippet:

```csharp
using NodaTime;
using System;

LocalDate start = new LocalDate(2017, 1, 1);
LocalDate end = new LocalDate(2017, 12, 31);

DateInterval value = new DateInterval(start, end);
value.Deconstruct(out LocalDate actualStart, out LocalDate actualEnd);

Console.WriteLine(actualStart);
Console.WriteLine(actualEnd);
```

I've pushed the changes now.